src/main/webapp/app/exam/participate/exam-navigation-sidebar/exam-navigation-sidebar.component.ts

```diff
@@ -0,0 +1,248 @@
+        if (!overviewPage) {
+            // out of index -> do nothing
+            if (exerciseIndex > this.exercises.length - 1 || exerciseIndex < 0) {
+                return;
+            }
+            // set index and emit event
+            this.exerciseIndex = exerciseIndex;
+            this.onPageChanged.emit({ overViewChange: false, exercise: this.exercises[this.exerciseIndex], forceSave: !!forceSave, submission: submission });
+        } else if (overviewPage) {
+            // set index and emit event
+            this.exerciseIndex = -1;
+            // save current exercise
+            this.onPageChanged.emit({ overViewChange: true, exercise: undefined, forceSave: false });
+        }
+        this.setExerciseButtonStatus(this.exerciseIndex);
+    }
+
+    /**
+     * Auxiliary method to call changeExerciseByIndex based on the unique id of the exercise
+     * @param exerciseId the unique identifier of an exercise that stays the same regardless of student exam ordering
+     */
+    changeExerciseById(exerciseId: number) {
+        const foundIndex = this.exercises.findIndex((exercise) => exercise.id === exerciseId);
+        this.changePage(false, foundIndex, true);
+    }
+
+    refreshExerciseSaveCount() {
+        this.numberOfSavedExercises = 0;
+        this.exercises.forEach((exercise) => {
+            const submission = ExamParticipationService.getSubmissionForExercise(exercise);
+            if (submission && submission.submitted) {
+                this.numberOfSavedExercises++;
+            }
+        });
+    }
+
+    /**
+     * calculate the exercise status (also see exam-exercise-overview-page.component.ts --> make sure the logic is consistent)
+     * also determines the used icon and its color
+     * TODO: we should try to extract a method for the common logic which avoids side effects (i.e. changing this.icon)
+     *  this method could e.g. return the sync status and the icon
+     *
+     * @param exerciseIndex index of the exercise
+     * @return the sync status of the exercise (whether the corresponding submission is saved on the server or not)
+     */
+    setExerciseButtonStatus(exerciseIndex: number): 'synced' | 'synced saved' | 'notSynced' {
+        this.icon = facSaveSuccess;
+        // If we are in the exam timeline we do not use not synced as not synced shows
+        // that the current submission is not saved which doesn't make sense in the timeline.
+        if (this.examTimeLineView) {
+            return this.exerciseIndex === exerciseIndex ? 'synced saved' : 'synced';
+        }
+
+        // start with a yellow status (save warning icon)
+        // TODO: it's a bit weird, that it works that multiple icons (one per exercise) are hold in the same instance variable of the component
+        //  we should definitely refactor this and e.g. use the same ExamExerciseOverviewItem as in exam-exercise-overview-page.component.ts !
```

Comment:
I'm not quite sure what this means. Could you clarify?

Reply:
This method is not written by me, so I am not sure who wrote that comment 😄 . I just gathered required methods from `exam-navigation-bar.component.ts` not to break any functionality of navigation.